src/metric/rank_metric.cc

```diff
@@ -1,21 +1,35 @@
 /*!
- * Copyright 2015 by Contributors
- * \file rank_metric.cc
- * \brief prediction rank based metrics.
- * \author Kailong Chen, Tianqi Chen
+ * Copyright 2020 XGBoost contributors
  */
+// When device ordinal is present, we would want to build the metrics on the GPU. It is possible
```

Comment:
This needs a little more thought from us if we want to use this approach going forward. Ideally we would have a consistent configuration logic when choosing GPU algorithms.